### DISCUSSION TITLE
[Beta] Minor issues/comments

### DISCUSSION THREAD

**Discussion by ryanbuening** [bluecaret/carettab#340] [Help and Support] [answered]
Beta is looking great! I noticed a few things outlined below:

1. Date widget - disabling the year in the date widget does not appear to remove the trailing comma.

![image](https://github.com/bluecaret/carettab/assets/812974/95e404e7-c61b-42d2-927b-3572deabef44)

2. Widget ids(?) are showing in settings

![image](https://github.com/bluecaret/carettab/assets/812974/adc15a2a-b3a6-4a5a-a27f-4dfb41f03658)

3. Weather widget - setting a font size of ~10 and enabling the "feels like" temp makes the "Feel like ..." text hard to read. Not sure if anything can be done here. Old version was a little more readable.

![image](https://github.com/bluecaret/carettab/assets/812974/02dc124c-aedf-4621-b882-9765c72ffbb3)

![image](https://github.com/bluecaret/carettab/assets/812974/8e234d6d-0538-457b-9617-6ec782f69464)

**Comment by bluecaret** [comment, marked as answer]
1) Good catch. I did have a reason for that originally, but since I coded that things were adjusted a bit and I don't think my reason makes sense anymore, so I'll fix that! thanks!

2) The IDs are on purpose. While developing I was noticing myself get confused when switching between different widgets of the same type. So to help users, by default the widget is named as the type followed by it's ID. You are given the option of renaming that widget using the text caret icon to the right. I wasn't sure what another option to help differentiate this would be. Just automatically going "1, 2, etc...) can get complicated in the code due to reordering/adding/removing widgets. If you have another idea here I'm open :)

3) I'll try adjusting the weather layouts a bit to ensure any particular text doesn't get too small hopefully. I'm not particularly pleased with the layouts I made to be honest, after your comment I might go back and at least add a "classic" layout that mimics the original closer.

Thanks!

**Comment by ryanbuening** [comment]
> The IDs are on purpose. [...]

Ahh...got it. 👍

---

The only other thing I noticed is that the app appears to be making a check license request to AWS on every load [here](https://github.com/bluecaret/carettab/blob/main/src/main.js#L44). You might be able to save a bunch of AWS calls by checking if the license exists in storage before making that call...just a thought.

![image](https://github.com/bluecaret/carettab/assets/812974/f048e071-1d96-4468-9ab8-1c6b49d407f0)

**Comment by bluecaret** [comment]
Ahh yes thank you! Actually I need to pull that out completely anyway. I had to make my own way to gift licenses to users, but after building that out the payment provider added the option in so I don't need that implementation after all :) So thanks for the reminder of it!

**Comment by ryanbuening** [comment]
Another thing I noticed is the text below will appear for a split second occasionally on load. I have noticed the beta version does seem to be loading a little slower than 3.x during a "cold-start" of Chrome...for me at least.

![image](https://github.com/bluecaret/carettab/assets/812974/25594693-0c65-47c2-bd2e-99bd795015cb)

**Comment by bluecaret** [comment]
Hey, just letting you know I've resolved everything mentioned here now. There won't be an update to the beta though, I think I'll be pushing to production in the next couple days. Thank you for your feedback!

In regards to the weather, I re-wrote the layouts for the widget. There are now the original two layouts available again, mostly as they were, and the other layouts have been adjusted.

**Reply by ryanbuening** [reply]
Awesome!
